pkg/lca/seedgenerator.go

```diff
@@ -0,0 +1,294 @@
+				Name: name,
+			},
+		},
+	}
+
+	if name == "" {
+		glog.V(100).Infof("The name of the seedgenerator is empty")
+
+		builder.errorMsg = "seedgenerator 'name' cannot be empty"
+	}
+
+	if !builder.ExistsSeedGenerator() {
+		return nil, fmt.Errorf("seedgenerator object %s doesn't exist", name)
+	}
+
+	builder.Definition = builder.Object
+
+	return &builder, nil
+}
+
+// DeleteSeedGenerator removes the existing seedgenerator from a cluster.
+func (builder *SeedGenerator) DeleteSeedGenerator() (*SeedGenerator, error) {
+	if valid, err := builder.validate(); !valid {
+		return builder, err
+	}
+
+	glog.V(100).Infof("Deleting the seedgenerator %s",
+		builder.Definition.Name)
+
+	if !builder.ExistsSeedGenerator() {
+		return builder, fmt.Errorf("seedgenerator cannot be deleted because it does not exist")
+	}
+
+	err := builder.apiClient.Delete(context.TODO(), builder.Definition)
+
+	if err != nil {
+		return builder, fmt.Errorf("can not delete seedgenerator: %w", err)
+	}
+
+	builder.Object = nil
+
+	return builder, nil
+}
+
+// GetSeedGenerator returns seedgenerator object if found.
+func (builder *SeedGenerator) GetSeedGenerator() (*lcasgv1alpha1.SeedGenerator, error) {
+	if valid, err := builder.validate(); !valid {
+		return nil, err
+	}
+
+	glog.V(100).Infof("Getting seedgenerator %s",
+		builder.Definition.Name)
+
+	seedgenerator := &lcasgv1alpha1.SeedGenerator{}
+	err := builder.apiClient.Get(context.TODO(), goclient.ObjectKey{
+		Name: builder.Definition.Name,
+	}, seedgenerator)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return seedgenerator, err
+}
+
+// ExistsSeedGenerator checks whether the given seedgenerator exists.
+func (builder *SeedGenerator) ExistsSeedGenerator() bool {
+	if valid, _ := builder.validate(); !valid {
+		return false
+	}
+
+	glog.V(100).Infof("Checking if seedgenerator %s exists",
+		builder.Definition.Name)
+
+	var err error
+	builder.Object, err = builder.GetSeedGenerator()
+
+	return err == nil || !k8serrors.IsNotFound(err)
+}
+
+// WithSeedImageSeedGenerator sets the seed image used by the seedgenerator.
+func (builder *SeedGenerator) WithSeedImageSeedGenerator(
+	seedImage string) *SeedGenerator {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	glog.V(100).Infof("Setting seed image %s in seedgenerator", seedImage)
+
+	builder.Definition.Spec.SeedImage = seedImage
+
+	return builder
+}
+
+// WithRecertImageSeedGenerator sets the recert image used by the seedgenerator.
+func (builder *SeedGenerator) WithRecertImageSeedGenerator(
+	recertImage string) *SeedGenerator {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	glog.V(100).Infof("Setting recert image %s in seedgenerator", recertImage)
+
+	builder.Definition.Spec.RecertImage = recertImage
+
+	return builder
+}
+
+// WaitUntilCompleteSeedGenerator waits the specified timeout for the seedgenerator to complete
+// actions.
+func (builder *SeedGenerator) WaitUntilCompleteSeedGenerator() (*SeedGenerator, error) {
```

Comment:
```suggestion
func (builder *SeedGenerator) WaitUntilComplete() (*SeedGenerator, error) {
```
Adding SeedGenerator to methods names is redundant

Reply:
Someone suggested we should do it to differentiate between 2 different components of the same package.

Reply:
That is only for functions

Reply:
i.e. lca pkg cannot have two functions called Pull

Reply:
Updated. Thanks.